### PR TITLE
Check if during pull profile an existing profile has the same username 

### PR DESCRIPTION
**Pull request by leventdem**

### CODE REVIEW

File: src/modals/QRCodeModal/QRCodeModal.jsx [hunk 1]
```diff
@@ -47,10 +47,16 @@ const QRCodeModal = ({ onClose, profile }) => {
...
-        } else if (e.message === 'alreadySynced') {
-          setMessage(t('This profile is already synchronized on this device.'))
+        switch (e.message) {
+          case 'abort':
+            setMessage(t('The sync process has been aborted, please try again.'))
+            break
+          case 'alreadySynced':
+            setMessage(t('This profile is already synchronized on this device.'))
+            break
+          case 'usernameAlreadyExists':
```
Comment (G-Ray) [inline src/modals/QRCodeModal/QRCodeModal.jsx:57]: ok

File: src/modals/QRCodeModal/QRCodeModal.jsx [hunk 2]
```diff
@@ -47,10 +47,16 @@ const QRCodeModal = ({ onClose, profile }) => {
...
+            setMessage(t('A profile with the same username already exists in this device, rename it and try again.'))
+            break
```
Comment (G-Ray) [inline src/modals/QRCodeModal/QRCodeModal.jsx:59]: we should set a "default" case

File: src/lib/sync-profile.js [hunk 3]
```diff
@@ -98,6 +98,19 @@ class SyncProfile {
       throw new Error('alreadySynced')
     }
 
+    const existingsProfiles = await this.masq.getProfiles()
+    const importedUsername = publicProfile.username.toUpperCase()
+    const usernameExist = existingsProfiles.find(_profile => _profile.username.toUpperCase() === importedUsername)
+
+    // HACK for tests. To test sync we create two dbs on the same device
+    if (usernameExist && process.env.NODE_ENV !== 'test') {
```
Comment (G-Ray) [inline src/lib/sync-profile.js:106]: ok, I can't think of a better solution as long as tests are run in the same browser's context

File: src/containers/Sync/Sync.jsx [hunk 4]
```diff
@@ -57,8 +58,13 @@ class Sync extends Component {
...
       this.setState({ message: 'error' })
       this.props.setSyncStep('error')
-      if (e.message === 'alreadySynced') {
-        this.setState({ message: t('This profile is already synchronized on this device.') })
+      switch (e.message) {
+        case 'alreadySynced':
+          this.setState({ message: t('This profile is already synchronized on this device.') })
+          break
+        case 'usernameAlreadyExists':
+          this.setState({ message: t('A profile with the same username already exists in this device, rename it and try again.') })
```
Comment (G-Ray) [inline src/containers/Sync/Sync.jsx:66]: need to handle a default case here too

File: src/modals/QRCodeModal/QRCodeModal.jsx (hunk 2 above)
Comment (leventdem) [reply src/modals/QRCodeModal/QRCodeModal.jsx:59]: ok done

File: src/containers/Sync/Sync.jsx (hunk 4 above)
Comment (leventdem) [reply src/containers/Sync/Sync.jsx:66]: ok done

**Review by G-Ray — APPROVED** [review]

**Comment by G-Ray** [comment]
We need to talk to the team to know if we need to compare uppercased username or not

**Review by G-Ray — APPROVED** [review]

**Pull request merged**